### PR TITLE
Update API samples and fix test issues

### PR DESCRIPTION
**Pull request by mhegazy**
Update the Samples to use new TS1.5 features like destructuring, let and const, and new es6 module syntax. also remove the content of typescript.d.ts from all baselines. should make test runs less noisy.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
Though please document the new harness flag.

File: src/harness/harness.ts [hunk 1]
```diff
@@ -781,7 +781,7 @@ module Harness {
 
             public reset() { this.fileCollection = {}; }
 
-            public toArray(): { fileName: string; file: WriterAggregator; }[] {
+            public toArray(): { fileName: string; file: WriterAggregator; }[]{
```
Comment (yuit) [inline src/harness/harness.ts]: Keep space before {

**Comment by yuit** [comment]
:+1:

**Pull request merged**